core/camel-core/src/test/java/org/apache/camel/processor/SagaTimeoutTest.java

```diff
@@ -64,6 +70,29 @@ public void testTimeoutHasNoEffectIfCompleted() throws Exception {
         compensate.assertIsNotSatisfied();
     }
 
+    @Test
+    public void testTimeoutMultiParticipants() throws Exception {
+
+        MockEndpoint compensate = getMockEndpoint("mock:compensate");
+        compensate.expectedMessageCount(1);
```

Comment:
That should be `2`, when a timeout expires all participants should be compensated

Reply:
There is only `service1` involved in the saga transction. `service2` can not join because the status is `COMPENSATED`. So I think it should be `1`.